lib/util/authorizer.js

```diff
@@ -123,8 +147,16 @@ class DefaultKeyResolver {
 	}
 }
 
+/**
+ * Resolves public keys from available resources
+ */
 module.exports = class Authorizer {
+	/**
+	 * Creates an instance of Authorizer.
+	 * @param {KeyResolverOptions} [options={}] Override default options
+	 */
 	constructor(options = {}) {
+		this._logger = options.logger
```

Comment:
How do you guarantee that _logger is set?  Is there an NPE risk here?

Reply:
There is minimal risk because of the following: 
https://github.com/SmartThingsCommunity/smartapp-sdk-nodejs/blob/master/lib/smart-app.js#L40
and
https://github.com/SmartThingsCommunity/smartapp-sdk-nodejs/blob/master/lib/util/log.js#L25-L36

_but_, I added a fallback `Log` instance in the updated code.

```js
this._logger = options.logger || new Log()
```

I also backed this with some unit tests. Not sure if they're elegant enough but it did seem to do the trick.